### PR TITLE
[REEF-1586] Fix CoreCLR issues in Org.Apache.REEF.Evaluator

### PR DESCRIPTION
**Pull request by rogancarr**
This addressed the issue by
  * Switching the use of Type.Assembly to TypeInfo.Assembly

JIRA:
  [REEF-1586](https://issues.apache.org/jira/browse/REEF-1586)

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.Evaluator/Evaluator.cs [hunk 1]
```diff
@@ -141,7 +142,7 @@ private static IConfiguration ReadEvaluatorConfiguration(string evaluatorConfigF
 
             AvroConfigurationSerializer serializer = new AvroConfigurationSerializer();
             var classHierarchy = TangFactory.GetTang()
-                .GetClassHierarchy(new string[] { typeof(ApplicationIdentifier).Assembly.GetName().Name });
+                .GetClassHierarchy(new string[] { typeof(ApplicationIdentifier).GetTypeInfo().Assembly.GetName().Name });
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Evaluator/Evaluator.cs]: What was the issue?

Comment (rogancarr) [inline lang/cs/Org.Apache.REEF.Evaluator/Evaluator.cs]: `Type.Assembly` is not supported in `.NET Core + Platform Extensions, Version=1.0`. The recommended solution is to access via `TypeInfo.Assembly`.

**Comment by rogancarr** [comment]
The [error](https://ci.appveyor.com/project/ApacheSoftwareFoundation/reef/build/1708-master/job/pyyaqdyih5itvioc#L10972) is in `VS2013`.

I'm not sure if this is a real error or transient.

```
"C:\projects\reef\lang\cs\TestRunner.proj" (default target) (1) ->
"C:\projects\reef\lang\cs\Org.Apache.REEF.Tests\Org.Apache.REEF.Tests.csproj" (Test target) (9) ->
(Test target) -> 
  C:\projects\reef\lang\cs\xunit.targets(52,5): error MSB3073: The command "..\packages\xunit.runner.console.2.1.0\tools\xunit.console.exe ..\bin\x64\Debug\Org.Apache.REEF.Tests\\Org.Apache.REEF.Tests.dll -notrait Environment=Yarn -html ..\bin\x64\Debug\Org.Apache.REEF.Tests\\xunit_report.html" exited with code 1. [C:\projects\reef\lang\cs\Org.Apache.REEF.Tests\Org.Apache.REEF.Tests.csproj]
```

File: lang/cs/Org.Apache.REEF.Evaluator/Evaluator.cs (hunk 1 above)
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Evaluator/Evaluator.cs]: You might want to add this into the JIRA description.

Comment (rogancarr) [inline lang/cs/Org.Apache.REEF.Evaluator/Evaluator.cs]: Oh, you are right. I've added this to the description.

**Comment by rogancarr** [comment]
Note that while this fixes the incompatibilities, the analyzer will still report an issue. More details here: https://issues.apache.org/jira/browse/REEF-1586

**Comment by jwang98052** [comment]
The test error is one of the transit.

File: lang/cs/Org.Apache.REEF.Evaluator/Evaluator.cs (hunk 1 above)
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Evaluator/Evaluator.cs]: Is this call supported on .NET 4.5?

Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Evaluator/Evaluator.cs]: I just answered my own question. The code compiles with .NET 4.5.1 as the target.

**Comment by markusweimer** [comment]
LGTM, will merge.

**Pull request closed without merging**